src/LibraBFT/Impl/Consensus/BlockStorage/Properties/BlockTree.agda

```diff
@@ -259,3 +253,94 @@ module insertQuorumCertESpec
+  Contract (Left x) = ⊤
+  Contract (Right (bt1 , il)) = ContractOk bt0 bt1 [] il
+
+  -- Here we prove the weakest precondition for insertQuorumCertE-AST to ensure Contract The proof
+  -- is similar to the corresponding one for insertQuorumCertE above, but the AST framework makes it
+  -- easier to discover the proof obligations.  Note that the code for insertQuorumCertE-AST is not
+  -- broken explicitly into steps because we don't need to write the types of the individual proofs
+  -- because the framework presents them to us.
+  contract-AST : ASTPredTrans.predTrans EitherPTExt insertQuorumCertE-AST Contract unit
+  contract-AST with safetyInvariant
+  --  The Left/bail cases are easy, as Contract simply requires ⊤ in this case.
+  ... | Left _ = tt
+  ... | Right unit =
+      -- Two proof obligations are introduced due to the use of maybeSD; the AST framework with the
+      -- branching extensions ensures that the two goals are created by typing C-c C-r in a hole
+      -- here.  To understand why, see the definition of maybeSD, which is in terms of the maybeD
+      -- field of the relevant MonadMaybeD instance (EitherDASTExt-MonadMaybeD), which translates to
+      -- ASTop (Right (BCmaybe mb)).  The opPT field of BranchPT establishes that we require two
+      -- proofs, one for the nothing case and one for the just case; each receives evidence that mb
+      -- is the relevant case (nothing or just something), and determines the relevant proof
+      -- obligation.
+      const tt , λ block _ →
+      -- Similarly, another use of maybeSD in the code results in automatically introducing two
+      -- proof obligations using C-c C-r
+      const tt ,
+      -- The "Right" goal in this case is determined by the use of ifAST in the code, which
+      -- translates to ASTop (Right (BCif b)) ... (see BranchingSyntax).  Therefore, in this case,
+      -- the proof obligation is determined in PredTransExtension to have two proof obligations, one
+      -- for the case in which b is true and one for b is false.  Each gets evidence of the value of
+      -- b (not used in this example) and the predicate transformer resulting from passing the value
+      -- of b to the function in the definition of ASTPredTrans.opPT yields the proof obligation.
+      -- The proofs for each case here are more or less copied from the insertQuorumcertESpec proof,
+      -- but because the framework guided us to the goals, we did not need to state their types
+      -- explicitly, which in turn means that we did not need to break the code into explicit steps.
+      λ _ _ →
+      (const $ let bt' = bt0 & btHighestCertifiedBlockId ∙~ block ^∙ ebId
+                             & btHighestQuorumCert       ∙~ qc
+                in ContractOk-trans
+                     (mkContractOk (∈BlockTree-upd-hqc refl refl))
+                     (contract-cont1' block bt' (fakeInfo ∷ [])))
+      , (const $ ContractOk-trans
```

Comment:
You have started a "alignment style" in this proof.  Might as well be consistent.  Meaning, align this:
```
(const $ ...
,
(const $ ...
```

Reply:
DONE